### ISSUE TITLE
False positive: Not all code paths return from a switch (if allowed values is a singleton-set)

### ISSUE THREAD

**Issue by hamsbrar** [vimeo/psalm#9910]
https://psalm.dev/r/06480444eb
```php
class A {
  const STATUS_FOO = 'foo';
}

/**
 * @param A::STATUS_* $s
 */
function foo(string $s) : string {
  switch ($s) {
    case A::STATUS_FOO:
      return 'hello';
  }
}
```
```
ERROR: [InvalidReturnType](https://psalm.dev/011) - 10:27 - Not all code paths of foo end in a return statement, return type string expected
```

Workaround: Directly using the literal `'foo'` in `case` seems to work ( https://psalm.dev/r/4d0f742d05 ).

**Comment by ygottschalk** [comment]
Having 2 class constants works: https://psalm.dev/r/c39020fe65
(In that case, inside a `default` case in the switch psalm does know that `$s` is typed `never`. In the original example, adding the same `default` case, psalm thinks `$s` is `'foo'` aka `A::STATUS_FOO`)

**Comment by orklah** [comment]
This is fixed :)
